presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/ShardCleaner.java

```diff
@@ -495,7 +495,10 @@ void cleanBackupShards()
             }
 
             try {
-                result.add(queue.poll(remainingNanos, NANOSECONDS));
+                T value = queue.poll(remainingNanos, NANOSECONDS);
+                if (value != null) {
```

Comment:
Do you think we should add a comment (because this only fails in Java 9)?

Reply:
This was always broken. The method is returning a collection with a null (which is valid for `ArrayList`) but then fails with `ConcurrentHashMap` (which never allowed null). So, while the code works in Java 8, it was never legal.